server/src/main/java/org/elasticsearch/monitor/jvm/DeadlockAnalyzer.java

```diff
@@ -137,7 +137,7 @@ public Deadlock(ThreadInfo[] members) {
             for (int x = 0; x < members.length; x++) {
                 ThreadInfo ti = members[x];
                 sb.append(ti.getThreadName());
-                if (x < members.length) {
+                if (x < members.length - 1) {
```

Comment:
Not sure about this. We're making a string of the form:

```threadName > ... > threadName > lockOwnerName```

This change removes the last `>` between the final `threadName` and the `lockOwnerName`. I think the original _behaviour_ looks right, although the condition was  always true so the `if` could be simply unwrapped.

Reply:
Sorry, yes I agree. I'll update that now.